packages/local_auth/local_auth_darwin/darwin/Tests/FLALocalAuthPluginTests.swift

```diff
@@ -0,0 +1,403 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import XCTest
+
+@testable import local_auth_darwin
+
+// Set a long timeout to avoid flake due to slow CI.
+let timeout: TimeInterval = 30.0
+
+/// A context factory that returns preset contexts.
+class StubAuthContextFactory: NSObject, FLADAuthContextFactory {
+  var contexts: [FLADAuthContext]
+  init(contexts: [FLADAuthContext]) {
+    self.contexts = contexts
+  }
+
+  func createAuthContext() -> FLADAuthContext {
+    XCTAssert(self.contexts.count > 0, "Insufficient test contexts provided")
+    let context = self.contexts.first!
+    contexts.remove(at: 0)
+    return context
+  }
+}
+
+class StubAuthContext: NSObject, FLADAuthContext {
+  /// Whether calls to this stub are expected to be for biometric authentication.
+  ///
+  /// While this object could be set up to return different values for different policies, in
+  /// practice only one policy is needed by any given test, so this just allows asserting that the
+  /// code is calling with the intended policy.
+  var expectBiometrics = false
+  /// The error to return from canEvaluatePolicy.
+  var canEvaluateError: NSError?
+  /// The value to return from evaluatePolicy:error:.
+  var evaluateResponse = false
+  /// The error to return from evaluatePolicy:error:.
+  var evaluateError: NSError?
+
+  // Overridden as read-write to allow stubbing.
+  var biometryType: LABiometryType = LABiometryType.none
+  var localizedFallbackTitle: String?
+
+  func canEvaluatePolicy(_ policy: LAPolicy) throws {
+    XCTAssertEqual(
+      policy,
+      expectBiometrics
+        ? LAPolicy.deviceOwnerAuthenticationWithBiometrics
+        : LAPolicy.deviceOwnerAuthentication)
+    if let canEvaluateError = canEvaluateError {
+      throw canEvaluateError
+    }
+  }
+
+  func evaluatePolicy(
+    _ policy: LAPolicy, localizedReason: String, reply: @escaping (Bool, Error?) -> Void
+  ) {
+    XCTAssertEqual(
+      policy,
+      expectBiometrics
+        ? LAPolicy.deviceOwnerAuthenticationWithBiometrics
+        : LAPolicy.deviceOwnerAuthentication)
+    // evaluatePolicy:localizedReason:reply: calls back on an internal queue, which is not
+    // guaranteed to be on the main thread. Ensure that's handled correctly by calling back on
+    // a background thread.
+    DispatchQueue.global(qos: .background).async {
+      reply(self.evaluateResponse, self.evaluateError)
+    }
+  }
+}
+
+// MARK: -
+
+class FLALocalAuthPluginTests: XCTestCase {
+
+  override func setUp() {
+    continueAfterFailure = false
```

Comment:
curious why? (also maybe add comment)

Reply:
No clue, it looks like it dates back to my adding the original test file. Copypasta from an XCUITest maybe? Removed.